Add remote Docker policy (#8312)
circleci/circleci-docs: jekyll/_cci2/remote-docker-images-support-policy.adoc

  - Cloud
  - Server v4.x
  - Server v3.x
---
= CircleCI remote Docker images support policy
:page-layout: classic-docs
:page-liquid:
:page-description: CircleCI remote Docker images release, update, and deprecation policy
:icons: font
:experimental:

[#overview]
== Overview

This document outlines the xref:building-docker-images#[CircleCI remote Docker image] release, update, and deprecation policy.

This policy applies to all CircleCI remote Docker images built for the remote Docker platform.

[#release-policy]
== Release policy

The CircleCI remote Docker images are based on our Linux VM images with Docker installed for the purposes of providing a remote environment that can execute Docker commands within jobs on the Docker executor.

We aim to support the latest two versions of the Docker Engine that are classified as within Security Support status.

Remote Docker images will be updated when a patch version is released from upstream. Tags will be redirected to the updated images automatically as described in the xref:#tagging[tagging] section of this document. We will announce these releases on our link:https://discuss.circleci.com/[Discuss Forum].

[#tagging]
== Tagging

> We support various tags for the remote Docker environment which allow customers to choose the type of remote Docker environment they require.

Reviewer: ```suggestion
We support various tags for the remote Docker environment to allow you to choose the type of remote Docker environment you require.
```

For the latest major version of Docker:

- `edge`: This image is for testing purposes. It is updated frequently without warning and could include breaking changes. 

- `current`: This image is the current stable image. This is the default if no tag is selected.

- `previous`: Once an `edge` image is promoted to current, the previous `current` image is moved to the `previous` tag.

> For the previous major version of Docker, we support a single tag following the format of `dockerXX` - e.g., `docker23` for Docker 23. This tag will point to the latest patch version of the major release, and will be updated if any patch versions are issued by the upstream.

Reviewer: ```suggestion
For the previous major version of Docker, we support a single tag following the format of `dockerXX`, for example, `docker23` for Docker 23. This tag will point to the latest patch version of the major release, and will be updated if any patch versions are issued by the upstream.
```

> For the previous major version of Docker, we support a single tag following the format of `dockerXX` - e.g., `docker23` for Docker 23. This tag will point to the latest patch version of the major release, and will be updated if any patch versions are issued by the upstream.

Reviewer: ```suggestion
For the previous major version of Docker, we support a single tag following the format of `dockerXX` - e.g., `docker23` for Docker 23. This tag will point to the latest patch version of the major release, and will be updated if any patch versions are issued upstream.
```

[#critical-cve-patches]
== Critical CVE patches

When critical CVEs are disclosed that affect the versions of the operating system or software stack in our remote Docker images, we will investigate the impact that this has on our images being used within the CircleCI execution environment. 

> In most cases, due to the ephemeral and isolated nature of the containers in the environment, it is not necessary to patch these images. We will always communicate our stance on these disclosures via our Discuss Forum.

Reviewer: ```suggestion
In most cases, due to the ephemeral and isolated nature of the containers in the environment, it is not necessary to patch these images. We will always communicate our stance on these disclosures on our link:https://discuss.circleci.com/[Discuss Forum].
```
